test/common/upstream/logical_dns_cluster_test.cc

```diff
@@ -115,7 +116,7 @@ TEST_P(LogicalDnsParamTest, ImmediateResolve) {
   )EOF";
 
   EXPECT_CALL(initialized_, ready());
-  EXPECT_CALL(dns_resolver_, resolve("foo.bar.com", std::get<1>(GetParam()), _))
+  EXPECT_CALL(*dns_resolver_, resolve("foo.bar.com", std::get<1>(GetParam()), _))
```

Comment:
Sorry for the radio silence - was busy with some other things. I think we should skip it (at least for now).

I just took a look at this yesterday and yeah, it's a bit of a pain to clutter the classes w/ the test getters, but also to cast from the high level Envoy virtual classes to their implementations. I think we're mostly covered in ensuring that the ares_channel gets the right setting by this test, https://github.com/lyft/envoy/pull/1018/files#diff-aafe7d77f5c31414857274dfc091160cR243. It's not end to end from configuration to the ares_channel, but we can add that later.